### PR TITLE
Add Stable Diffusion 3 guide for KerasHub

### PR DESCRIPTION
**Pull request by james77777778**
@divyashreepathihalli 

This guide covers all the APIs for SD3 in KerasHub. Please let me know if any changes are required.

### CODE REVIEW

**Review by divyashreepathihalli — COMMENTED** [review]
Thanks for adding this super cool guide @james77777778!! I have left a few comments.

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+Note that in this guide, we'll use `height=512` and `width=512` for faster
+image generation. For higher-quality output, it's recommended to use the default
+size of `1024`. Since the entire backbone has about 3 billion parameters, which
+can be challenging to fit into a consumer-level GPU, we set `dtype="float16"` to
+reduce the usage of GPU memory -- the officially released weights are also in
+float16.
+
+It is also worth noting that the preset "stable_diffusion_3_medium" excludes the
+T5XXL text encoder, as it requires significantly more GPU memory. The performace
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py:74]: I've added this line:
```
The weights, including T5XXL, will be available on KerasHub soon.
```

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+
+"""
+## Text-to-image task
+
+Now we know the basis of the Stable Diffusion 3 and the text-to-image task.
+Let's explore it more by KerasHub APIs.
+
+To enable batch processing, we can feed a list of prompts into the model:
+"""
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py:173]: Changed to:

```python
def display_generated_images(images):
    """Helper function to display the images from the inputs.

    This function accepts the following input formats:
    - 3D numpy array.
    - 4D numpy array: concatenated horizontally.
    - List of 3D numpy arrays: concatenated horizontally.
    """
    display_image = None
    if isinstance(images, np.ndarray):
        if images.ndim == 3:
            display_image = Image.fromarray(images)
        elif images.ndim == 4:
            concated_images = np.concatenate(list(images), axis=1)
            display_image = Image.fromarray(concated_images)
    elif isinstance(images, list):
        concated_images = np.concatenate(images, axis=1)
        display_image = Image.fromarray(concated_images)

    if display_image is None:
        raise ValueError("Unsupported input format.")

    plt.figure(figsize=(10, 10))
    plt.axis("off")
    plt.imshow(display_image)
    plt.show()
    plt.close()
```

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+styles and elements. The input format becomes a dict with the keys `"prompts"`
+and `"negative_prompts"`.
+
+If `"negative_prompts"` is not provided, it will be interpreted as an
+unconditioned prompt with the default value of `""`.
+"""
+
+inputs = {"prompts": [prompt] * 3, "negative_prompts": ["Green color"] * 3}
+generated_images = text_to_image.generate(inputs)
+generated_image = concate_images(generated_images)
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py]: Please see the above comment that a helper function has been added.

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+The input format becomes a dict with the keys `"images"`, `"prompts"` and
+optionally `"negative_prompts"`.
+"""
+
+image_to_image = keras_hub.models.StableDiffusion3ImageToImage(backbone, preprocessor)
+
+image = keras.utils.get_file(
+    "cat.png",
+    origin="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/cat.png",
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py]: I was unable to verify the copyright of that image, so I replaced it with `mountain_dog.jpeg` and manually annotated a mask.

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+image_array = image_array / 127.5 - 1.0
+prompt = "cat wizard, gandalf, lord of the rings, detailed, fantasy, cute, "
+prompt += "adorable, Pixar, Disney, 8k"
+
+generated_image = image_to_image.generate({"images": image_array, "prompts": prompt})
+
+display_image = concate_images([np.array(image), generated_image])
+plt.axis("off")
+plt.imshow(display_image)
+
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py:307]: Added:
```
As you can see, a new image is generated based on the reference image and the
prompt.
```

File: guides/keras_hub/stable_diffusion_3_in_keras_hub.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+generated_image = inpaint.generate(
+    {"images": image_array, "masks": mask_array, "prompts": prompt}
+)
+
+display_image = concate_images(
+    [np.array(image), np.array(mask.convert("RGB")), generated_image]
+)
+plt.axis("off")
+plt.imshow(display_image)
+
```
[the remark this answers is not in the document]
Comment (james77777778) [reply guides/keras_hub/stable_diffusion_3_in_keras_hub.py:356]: Added:
```
Fantastic! The dog is replaced by a cute black cat, but unlike image-to-image,
the background is preserved.
```

**Comment by james77777778** [comment]
@divyashreepathihalli 
Thank you for the detailed review. The comments should now be addressed.

**Review by divyashreepathihalli — APPROVED** [review]
Thanks Hongyu! LGTM

**Review by fchollet — APPROVED** [review]
